src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -1823,15 +1824,34 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
                         return text.Insert(text.LastIndexOf(".") + 1, "\n");
                     return text;
                 case "ClassButton":
-                    text = Dynamo.Nodes.Utilities.InsertSpacesToString(text);
-                    if (text.Length > Configurations.MaxLengthRowClassButtonTitle)
+                    string originalName = Dynamo.Nodes.Utilities.InsertSpacesToString(text);
+                    text = originalName;
+                    int maxRowLength = Configurations.MaxLengthRowClassButtonTitle;
+                    if (text.Length > maxRowLength)
                     {
+                        // If clas name is like "Cooooooool Name", then this truncation is wrong.
                         if (text.IndexOf(" ") != -1)
```

Comment:
`text.IndexOf(" ")` getting called multiple times when they can be stored in a variable.